src/solver/optimize.cpp

```diff
@@ -1,14 +1,294 @@
+#include <math.h>
+#include <string>  
+#include <iostream> 
+#include <sstream> 
 #include "../components/token.hpp"
 #include "../components/order.hpp"
 #include "../components/amm.hpp"
 #include "./optimize.hpp"
+#include "../../external_libs/json.hpp"
 
-void solve_auction(std::vector<Token> &tokens, std::vector<Order> &orders, std::vector<CP_AMM> &amms)
+
+// auxiliary function (used temporarily) for computing x^n using the boost library data types
```

Comment:
[It seems](https://www.boost.org/doc/libs/1_78_0/libs/multiprecision/doc/html/boost_multiprecision/ref/number.html) there is a power function in the library. Is there any reason why you're coding your own?

Reply:
Good point. Will look into it. I had just done this as a temporary fix and forgot to replace it.

Reply:
Seems that I was actually not using after some point. For now, the standard pow() function of C++ is used. Will test if that is enough or not.